File: 2873-maximum-value-of-an-ordered-triplet-i/2873-maximum-value-of-an-ordered-triplet-i.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumTripletValue = function(nums) {
    
    let result = 0;
    let length = nums.length;

    for(let i = 0 ; i < length - 2 ; i++)
    {
        for(let j = i + 1 ; j < length - 1 ; j++)
        {
            for(let k = j + 1 ; k < length ; k++)
            {
                let val = (nums[i] - nums[j]) * nums[k];
                result = Math.max(val, result);
            }
        }
    }

    return result;
};

File: 2873-maximum-value-of-an-ordered-triplet-i/2873-maximum-value-of-an-ordered-triplet-i.java
class Solution {
    public long maximumTripletValue(int[] nums) {
        
        int length = nums.length;
        long result = 0;
        
        for(int i = 0 ; i < length - 2 ; i++)
        {
            for(int j = i + 1 ; j < length - 1 ; j++)
            {
                for(int k = j + 1 ; k < length ; k++)
                {
                    long val = (long)(nums[i] - nums[j]) * (long)nums[k];
                    result = Math.max(result, val);
                }
            }
        }

        return result;
    }
}